examples/svm/plot_oneclass.py

```diff
@@ -37,24 +35,52 @@
 n_error_test = y_pred_test[y_pred_test == -1].size
 n_error_outliers = y_pred_outliers[y_pred_outliers == 1].size
 
-# plot the line, the points, and the nearest vectors to the plane
-Z = clf.decision_function(np.c_[xx.ravel(), yy.ravel()])
-Z = Z.reshape(xx.shape)
+# %%
+import matplotlib.font_manager
+import matplotlib.lines as mlines
+import matplotlib.pyplot as plt
+
+from sklearn.inspection import DecisionBoundaryDisplay
 
-plt.title("Novelty Detection")
-plt.contourf(xx, yy, Z, levels=np.linspace(Z.min(), 0, 7), cmap=plt.cm.PuBu)
-a = plt.contour(xx, yy, Z, levels=[0], linewidths=2, colors="darkred")
-plt.contourf(xx, yy, Z, levels=[0, Z.max()], colors="palevioletred")
+_, ax = plt.subplots()
+
+# generate grid for the boundary display
+xx, yy = np.meshgrid(np.linspace(-5, 5, 500), np.linspace(-5, 5, 500))
```

Comment:
Just a tweak. We don't really need a dense grid with `DecisionBoundaryDisplay`.
```suggestion
xx, yy = np.meshgrid(np.linspace(-5, 5, 10), np.linspace(-5, 5, 10))
```

Reply:
It looks like we should have a `xlim` and `ylim` to be able to overwrite the limits of the grid. This is a bit annoying to generate the grid and dataset outside.

Reply:
I added this option in https://github.com/scikit-learn/scikit-learn/issues/27462